nomad/structs/structs.go

```diff
@@ -1965,6 +1940,14 @@ func (n *Node) Canonicalize() {
 			}
 		}
 	}
+
```

Comment:
this is superfluous from the point of view of upgrading... however, there were some tests in the test suite that set `DrainStrategy` but didn't bother to set `SchedulingEligibility`, so this covered for them. also, never pass up a chance to enforce invariants.

Reply:
I think this is necessary, and I actually left a question about where it lives on the removal above before scrolling down! :sweat_smile: 

We need to ensure the invariant  `SchedulingEligibility==Ineligible when DrainStrategy!=nil` is enforced on all node updates from the CLI/HTTP-API/RPC. Canonicalize is our usual place of enforcing such constraints, so this seem right to me. Maybe add a comment along the lines of:

> Ensure SchedulingEligibility is set whenever draining so the plan applier and other scheduling logic only need to check SchedulingEligibility when determining whether a placement is feasible on a node.